src/core/thread/network_data_local.cpp

```diff
@@ -66,7 +66,7 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
 
 Error Local::RemoveOnMeshPrefix(const Ip6::Prefix &aPrefix)
 {
-    return RemovePrefix(aPrefix, NetworkDataTlv::kTypeBorderRouter);
+    return RemovePrefix(aPrefix);
```

Comment:
We can consider making this and the `RemoveHasRoutePrefix()` defined as inline (since only calling the other method)?